packages/lit-dev-content/src/components/litdev-github-share-button.ts

```diff
@@ -32,20 +48,50 @@ export class LitDevGitHubShareButton extends LitElement {
   @property()
   authorizeUrl?: string;
 
+  /**
+   * Base URL for the GitHub API.
+   */
+  @property()
+  githubApiUrl?: string;
+
+  /**
+   * A function to allow this component to access the project upon save.
+   */
+  getProjectFiles?: () => SampleFile[] | undefined;
+
   override render() {
     return html`<button @click=${this._onClick}>Share with GitHub</button>`;
   }
 
   private async _onClick() {
-    if (!this.clientId || !this.authorizeUrl) {
+    const projectFiles = this.getProjectFiles?.();
```

Comment:
I manually tested deleting all files and then sharing an empty gist. When I loaded the gist it said "Playground file not found" but loaded a package.json. Can this ever be a falsey value?
No action required.

Reply:
"Playground not found" is what the Playground service worker serves when there is no index.html in the project. Filed https://github.com/google/playground-elements/issues/235 to track making a nicer message there.

Interesting find about the package.json -- looks like playground has a bug where if you load a project that has no visible files, but does have a hidden file, then hidden file will be displayed. Filed https://github.com/google/playground-elements/issues/234.

The reason there is a package.json is there is that we automatically add our default one if there wasn't one already in the project when you load a project.

Also GitHub does not allow creating a gist with no files in it. Added a check for that, with a TODO to make the button disabled (slightly tricky in the current factoring, because we only fetch the project files on-demand when you save).

Reply:
Also fixed a bug where I was taking the wrong URL parameter for the base64 case.